src/stdlib_math_is_close.fypp

```diff
@@ -0,0 +1,43 @@
+#:include "common.fypp"
+
+submodule(stdlib_math) stdlib_math_is_close
+
+    use, intrinsic :: ieee_arithmetic, only: ieee_is_nan
+    implicit none
+    
+contains
+
+    #! Determines whether the values of `a` and `b` are close.
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    elemental module logical function is_close_${t1[0]}$${k1}$(a, b, rel_tol, abs_tol, equal_nan) result(close)
+        ${t1}$, intent(in) :: a, b
+        real(${k1}$), intent(in), optional :: rel_tol, abs_tol
+        logical, intent(in), optional :: equal_nan
+        logical :: equal_nan_
+
+        equal_nan_ = optval(equal_nan, .false.)
+        
+        if (ieee_is_nan(a) .or. ieee_is_nan(b)) then
+            close = merge(.true., .false., equal_nan_ .and. ieee_is_nan(a) .and. ieee_is_nan(b))
+        else
+            close = abs(a - b) <= max( abs(optval(rel_tol, 1.0e-9_${k1}$)*max(abs(a), abs(b))), &
+                                       abs(optval(abs_tol, 0.0_${k1}$)) )
+        end if     
```

Comment:
Here is the internal core business of `**_close`

Reply:
The relative tolerance is kind of an arbitrary choice here, could we derive it from something like `sqrt(epsilon(...))` (should be around 1.0e-9 for `0.0_dp`).

Reply:
`1.0e-9` here is the default relative tolerance value from the python standard library: https://docs.python.org/3.9/library/math.html#math.isclose
```python
math.isclose(a, b, *, rel_tol=1e-09, abs_tol=0.0)
```

Using an exact value `1.0e-9` may help users easy-to-remember.

Reply:
Should this value depend on the kind of `real`. E.g.
```fortran
print*,epsilon(1.)
end
```
returns 
```
  1.1920929E-07
```
with ifort 2021 on my computer.

`sqrt(epsilon(..))` sounds therefore a good suggestion (or a defined value close to this result, and provided in the spec).

Reply:
```fortran
print '(*(es0.4,2x))', epsilon(1.0), sqrt(epsilon(1.0))
print '(*(es0.4,2x))', epsilon(1.0d0), sqrt(epsilon(1.0d0))
print '(*(es0.4,2x))', epsilon(1.0_16), sqrt(epsilon(1.0_16))
end
```
Results:
```sh
1.1921E-7  3.4527E-4
2.2204E-16  1.4901E-8
1.9259E-34  1.3878E-17
```
It seems that `1.0e-9` is a compromise choice, and using `sqrt(epsilon(..))` also seems good.